homeassistant/components/elmax/__init__.py

```diff
@@ -0,0 +1,284 @@
+"""The elmax-cloud integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+from logging import Logger
+from typing import Any, Mapping
+
+import async_timeout
+from elmax_api.exceptions import (
+    ElmaxApiError,
+    ElmaxBadLoginError,
+    ElmaxBadPinError,
+    ElmaxNetworkError,
+)
+from elmax_api.http import Elmax
+from elmax_api.model.endpoint import DeviceEndpoint
+from elmax_api.model.panel import PanelEntry, PanelStatus
+
+from homeassistant.components.elmax.const import (
+    CONF_ELMAX_PANEL_ID,
+    CONF_ELMAX_PANEL_PIN,
+    CONF_ELMAX_PASSWORD,
+    CONF_ELMAX_USERNAME,
+    DEFAULT_TIMEOUT,
+    DOMAIN,
+    ELMAX_PLATFORMS,
+    POLLING_SECONDS,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.exceptions import ConfigEntryAuthFailed, HomeAssistantError
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.typing import HomeAssistantType
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistantType, entry: ConfigEntry) -> bool:
+    """Set up elmax-cloud from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
+
+    # Retrieve username/password from integration configuration
+    username: str = entry.data[CONF_ELMAX_USERNAME]
+    password: str = entry.data[CONF_ELMAX_PASSWORD]
+    panel_id: str = entry.data[CONF_ELMAX_PANEL_ID]
+    panel_pin: str = entry.data[CONF_ELMAX_PANEL_PIN]
```

Comment:
These seem to be used only to create the `ElmaxCoordinator` so there is no need to create intermediate variables.

Reply:
Not really, I also use them for schema validation in `config_flow`. Now that Elmax coordinator has been moved as per your suggestion in a dedicated class, these are now legit.

Reply:
Well yes, you dont need them

Reply:
Sorry I misunderstood your point (I thought you were asking to remove CONF_ELMAX_ constants, not the variables). That is totally correct!